Added a guide for how to add a new platform icon to the docs (#9894)
getsentry/sentry-docs: docs/contributing/platforms/platform-icons/index.mdx

---
title: Platform Icons
sidebar_order: 1
---

> To have consistent icons for all supported platforms over all of Sentry (the product, the docs, landing pages, etc) we created a repo for managing those icons, it is called [platformicons](https://github.com/getsentry/platformicons).

Reviewer: ```suggestion
To have consistent icons for all supported platforms across all of Sentry (the product, the docs, landing pages, etc.) we created a repo for managing those icons, called [platformicons](https://github.com/getsentry/platformicons).
```

If there is a new platform we support, there are several steps to be done, so the new icon can be used in the docs:

1. **Add the new icon to `platformicons`:**

>     Create a [new issue on the platformicons repository](https://github.com/getsentry/platformicons/issues) and specify how the icon should be named and a link to the website of the platform you want to add. Also, make sure to describe whether it is a top level icon (like `python`) or a subitem that is only available in one specific platform (like `python.celery`).

Reviewer: ```suggestion
    Create a [new issue in the platformicons repository](https://github.com/getsentry/platformicons/issues) that specifies how the icon should be named and includes a link to the platform's website. Also, make sure to specify whether it is a top-level icon (like `python`) or a subitem that is only available in one specific platform (like `python.celery`).
```

>     You can also create a PR in the repository where you add two versions of the icon. See [README.md](https://github.com/getsentry/platformicons) for the format.

Reviewer: ```suggestion
    You can also create a PR in the `platformicons` repository where you add two versions of the icon. See the [README](https://github.com/getsentry/platformicons) for more info on formatting.```

2. **Release a new version of `platformicons` to NPM.**

>     If you are a Sentry employee follow the [README.md](https://github.com/getsentry/platformicons) on how to do a release.

Reviewer: ```suggestion
    If you are a Sentry employee, follow the [README.md](https://github.com/getsentry/platformicons) on how to do a release.
```

>     If you are not a Sentry employee ask on our [Discord server](https://discord.com/invite/Ww9hbqr) so someone can review your contribution and trigger the release.

Reviewer: ```suggestion
    If you are not a Sentry employee, ask on our [Discord server](https://discord.com/invite/Ww9hbqr) so someone can review your contribution and trigger the release.
```

> 3. **Update the `platformicons` version used in docs repository.**

Reviewer: ```suggestion
3. **Update the `platformicons` version used in the docs repository.**
```

    Update the version of your newly released `platformicons` in `package.json` and run `yarn install` to also have the new version in `yarn.lock`.

4. **Make it available as a component in the docs repository.**

>     Update [platformIcon.tsx](https://github.com/getsentry/sentry-docs/blob/master/src/components/platformIcon.tsx) to have your new icon.
> 
>     You need to:
>     - add two imports (one for each version of the icon)
>     - add it to `formatToSVG`
>     - add it to `PLATFORM_TO_ICON`

Reviewer: ```suggestion
    Update [platformIcon.tsx](https://github.com/getsentry/sentry-docs/blob/master/src/components/platformIcon.tsx) to include your new icon. Make sure to:

    - Add two imports (one for each version of the icon).
    - Add the icons to `formatToSVG`.
    - Add a mapping for the platform and icon to `PLATFORM_TO_ICON`.
```

> 4. **Now the new Icon is available in the docs and will be used wherever icons are rendered!**

Reviewer: ```suggestion
Now the new icon is available in the docs and will be used wherever icons are rendered!
```
Not really a step, so let's make this its own paragraph.
